### DISCUSSION TITLE
Why some devices report battery OK and others don't ?

### DISCUSSION THREAD

**Discussion by kevincaradant** [Koenkk/zigbee2mqtt#5532] [Devices]
Hi 

In the UI I have only 3 or 4 devices with the green icon.

And it's weird but I have one device green and 2 others black but it's the same device and same version

![image](https://user-images.githubusercontent.com/6093175/103487368-1a6baf80-4e05-11eb-9042-d686bcead0e7.png)

Any idea how is it working ? 

Thank you :)

**Comment by Koenkk** [comment]
@nurikk can you explain this?

**Comment by kevincaradant** [comment]
To be honest it's the first time I see that we can have something else than black. Just because I paired two new devices which have the green battery as exepected I guess.

So I decided to look others (68 devices) in the list and it's at this moment that I saw only one other green in the list (between 2 others in the same reference always black). It's at this moment where I was a little bit confused about color :/

![image](https://user-images.githubusercontent.com/6093175/103488865-869fe080-4e10-11eb-8842-1c197135cc40.png)

For information, when I hover the device (battery green), I have: `Battery OK, power level 99%`
When I hover the device (battery black), I have `Battery, power level 96%`. So I don't have the `OK` ?

Is it a bug maybe ?

**Comment by nurikk** [comment]
Some devices are using battery_ok and battery values, green colour reflects ok=true. Have a look at source code for next time

**Reply by kevincaradant** [reply]
Yes I understood that color was reflected by OK = true but that  don't explain why some devices have it and some others doesn't have it. Specially if it's the SAME device  type (looks like a bug in this case).

I prefer to repeat what I said because not sure you see it ')
Look my picture in the previous message (2sd line)
